client/src/components/DeskList.tsx

```diff
@@ -0,0 +1,39 @@
+import { List, ListItemButton } from "@mui/material";
+import { useState, FC, useEffect, useContext } from "react";
+import { FormContext } from "../pages/bookingPage";
+import { DeskType } from "../types";
+
+function DeskItem() {
```

Comment:
This function returns an array, so it's not a DeskItem but DeskItems. I think in this specific case you might as well make this the DeskList component. 

If you really want to split it up, you can do it like this:

```
// In DeskList.tsx
export const DeskList: FC = () => {
  const [currentDeskList ...
  return <List>{currentDeskList.map(desk => <DeskItem desk={desk} />}</List>
}

// In DeskItem.tsx
export const DeskItem: FC<Props> = ({ desk }) => {
  return <ListItemButton ...
}
```